proposals/004_Overlays.md

```diff
@@ -0,0 +1,194 @@
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="overlayVersion"></a>overlay | `string` | Version of the Overlay specification that this document conforms to. 
+<a name="overlayInfo"></a>info | [[Info Object](#overlayInfoObject)] | Identifying information about the overlay.
+<a name="overlayExtends"></a>extends | `url` | URL to an OpenAPI document this overlay applies to. 
+<a name="overlayUpdates"></a>updates | [[Update Object](#updateObject)] | A list of update objects to be applied to the target document.
+
+The list of update objects MUST be applied in sequential order to ensure a consistent outcome.  Updates are applied to the result of the previous updates. This enables objects to be deleted in one update and then re-created in a subsequent update.
+
+The `extends` property can be used to indicate that the Overlay was designed to update a specific OpenAPI description.  This is an optional property.  Where no `extends` is provided it is the responsibility of tooling to apply the Overlay documents to the appropriate OpenAPI description.
+
+#### <a name="overlayInfoObject"></a>Info Object
+
+This object contains identifying information about the [OpenAPI Overlay document](#oasDocument).
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="overlayTitle"></a>title | `string` | A human readable description of the purpose of the overlay.
+<a name="overlayVersion"></a>version | `string` | A version identifer for indicating changes to an overlay document.
+
+#### <a name="updateObject"></a>Update Object
+
+This object represents one or more changes to be applied to the target document at the location defined by the target JMESPath.
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="updateTarget"></a>target | `string` | A JMESPath expression referencing the target objects in the target document.
+<a name="updateValue"></a>value | [Any](#valueObject) | An object with the properties and values to be updated in the target document.  Property has no impact if `remove` property is `true`.
+<a name="updateRemove"></a>remove | `boolean` | A boolean value that indicates that the target object is to be removed from the the map or array it is contained in. The default value is false.  
+
+The properties of the `Value Object` MUST be compatible with the target object referenced by the JMESPath key.  When the Overlay document is applied, the properties in the `Value Object` replace properties in the target object with the same name and new properties are appended to the target object.
+
+##### Structured Overlays Example
+
+When updating properties throughout the target document it may be more efficient to create a single `Update Object` that mirrors the structure of the target document. e.g.
+
+```yaml
+overlay: 1.0.0
+info:
+  title: Structured Overlay
+  version: 1.0.0
+updates:
+- target: "@"
+    value:
```

Comment:
Bad indentation (it's repeated on every example).

Reply:
Sequences (aka arrays) don't need to be indented https://yaml.org/spec/1.2/spec.html#id2802662

> Since people perceive the “-” indicator as indentation, nested block sequences may be indented by one less space to compensate
https://yaml.org/spec/1.2/spec.html#id2799181

Reply:
Mmm, I didn't mean the `- target` but the indentation on `value`.

Reply:
Dammit. Sorry about that.  I'll fix it.